src/main/scala/nl/knaw/dans/easy/fedoratobag/EasyFedoraToBagApp.scala

```diff
@@ -219,55 +219,45 @@ class EasyFedoraToBagApp(configuration: Configuration) extends DebugEnhancedLogg
         .getOrElse(Success(()))
       _ <- managedMetadataStream(foXml, "DATASET_LICENSE", bag, "depositor-info/depositor-agreement")
         .getOrElse(Success(()))
+      isOriginalVersioned = options.transformationType == ORIGINAL_VERSIONED
```

Comment:
`bag` is a parameter for several method calls.  You could define it as an `implicit` variable outside for comprehension.

Reply:
I would rename the following methods as follows:
addXmlMetadataTo -> addXmlMetadataToBag
addMetadataStreamTo -> addMetadataStreamToBag
addAgreementsTo ->addAgreementsToBag

Reply:
Adding bag to the method names would make the "phrases" stammer: `...Bag(bag, ...`, both for the definition as for the calls.

Reply:
I am not sure if I understand what you mean.  But if you mean that you have  `bag` both in the method name and as a parameter, this would not be the case if you would make the `bag` parameter `implicit` as I suggest above...

Reply:
That might become very confusing and dangerous as we might be dealing with two bags for a single dataset. Fixed it though for the nested method `fillSecondBag.copy`. 

Even if the conflict could be avoided in the current state of the code, it might cause problems in the future. Creating the bag before the long for comprehension of `createBag` would also mean to create them when not needed at all.